iwyu_tool.py

```diff
@@ -151,6 +151,12 @@ def main(compilation_db_path, source_files, verbose, formatter, iwyu_args):
                 print('WARNING: \'%s\' not found in compilation database.' %
                       source)
 
+    # Remove entries with directory belonging to any ignore_dirs
+    def is_ignored(dirpath):
+        return any(dirpath.startswith(os.path.realpath(i)) for i in ignore_dirs)
```

Comment:
Let's also hoist the `realpath` call;

      ignore_dirs = [os.path.realpath(i) for i in ignore_dirs

      def is_ignored(dirpath):
          return any(dirpath.startswith(i) for i in ignore_dirs)